src/ameba.ts

```diff
@@ -29,13 +29,24 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument): void {
-        if (document.languageId !== 'crystal' || document.isUntitled || document.uri.scheme !== 'file') {
-            return;
+    public execute(document: TextDocument, virtual: boolean = false): void {
+        if (!isCrystalDocument(document)) return;
+        if (isDocumentVirtual(document) && !virtual) return;
+
+        const dir = (workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document.uri)).uri.fsPath;
+
+        const args = [this.config.command, '--format', 'json'];
+
+        if (!virtual) {
+            args.push(document.fileName)
+        } else {
+            // Indicate that the source is passed through STDIN
+            args.push('-');
+
+            // Disabling these as they're common when typing
+            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace,Naming/Filename');
```

Comment:
No need to disable this one anymore.

```suggestion
            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace');
```